packages/core/bootstrap/src/lib/config/provider-limits/config.ts

```diff
@@ -30,8 +48,35 @@ export function get(
   const perMinuteRateLimit = getEnv('RATE_LIMIT_CAPACITY_MINUTE')
   const shouldIgnorePerSecLimit = perSecRateLimit && parseInt(perSecRateLimit) <= 0
   const shouldIgnorePerMinLimit = perMinuteRateLimit && parseInt(perMinuteRateLimit) <= 0
-  if (perSecRateLimit) capacity = shouldIgnorePerSecLimit ? 0 : parseInt(perSecRateLimit)
-  if (perMinuteRateLimit) capacity = shouldIgnorePerMinLimit ? 0 : parseInt(perMinuteRateLimit)
+  let highestTierLimit = 0
+  let lastTier: string | undefined = undefined
+
+  if (!capacity && enabled) {
+    lastTier = getLastTierLimit(rateLimitConfig)
+  }
+
+  if (perSecRateLimit) {
+    capacity = shouldIgnorePerSecLimit ? 0 : parseInt(perSecRateLimit)
+    if (lastTier)
+      highestTierLimit = shouldIgnorePerSecLimit
+        ? 0
+        : (rateLimitConfig.limits.http[lastTier].rateLimit1m as number)
+  }
+
+  if (perMinuteRateLimit) {
+    capacity = shouldIgnorePerMinLimit ? 0 : parseInt(perMinuteRateLimit)
+    if (lastTier)
+      highestTierLimit = shouldIgnorePerSecLimit
+        ? 0
+        : (rateLimitConfig.limits.http[lastTier].rateLimit1m as number)
+  }
```

Comment:
This logic seems too convoluted. Can we just modify `getLastTierLimit` to simply return the highest number?